jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -380,31 +364,46 @@ public Iterator<MultiPart.Part> iterator()
         {
             return parts.iterator();
         }
+    }
 
-        /**
-         * <p>Returns a new {@link MultiPart.ContentSource} with the same boundary
-         * as this object, and containing all the parts contained in this object.</p>
-         *
-         * @return a new {@link MultiPart.ContentSource} with all the parts of this object
-         */
-        public MultiPart.ContentSource toContentSource()
+    /**
+     * <p>The multipart/form-data specific content source.</p>
+     *
+     * @see MultiPart.AbstractContentSource
+     */
+    public static class ContentSource extends MultiPart.AbstractContentSource
+    {
+        public ContentSource(String boundary)
         {
-            MultiPart.ContentSource result = new MultiPart.ContentSource(getBoundary());
-            parts.forEach(result::addPart);
-            result.close();
-            return result;
+            super(boundary);
+        }
+
+        protected HttpFields customizePartHeaders(MultiPart.Part part)
+        {
+            HttpFields headers = part.getHeaders();
```

Comment:
I think the following makes it clearer that this is extending the existing behaviour:
```suggestion
            HttpFields headers = super.customizePartHeaders(part);
```